### PR TITLE
CC-24280 Jenkins migration 1.9.6

### PR DESCRIPTION
**Pull request by cprovencher**
### Why are we migrating?
[Read the background and motivation](https://confluentinc.atlassian.net/wiki/spaces/TOOLS/pages/3033867557/CI+System+Migration+Jenkins-+Semaphore#Background-and-Motivation)  
Please merge this PR ASAP because we plan on shutting down Jenkins by the end of the year.

### CODE REVIEW

**Comment by akanimesh7** [comment]
@cprovencher 
1. Why didn't we remove the Jenkinsfile
2. the branches on which we work are of the format -- `v\d+\.\d+\.\d+-hotfix-x`. Can you whitelist these ? For ex -- v1.9.6-hotfix-x

File: service.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: debezium
+lang: java
+lang_version: 11
```
Comment (rajdangwal) [inline service.yml:3]: Again, should this be 17?

**Review by rajdangwal — COMMENTED** [review]
Thanks @akanimesh7. Left some comments.

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      epilogue:
+        always:
+          commands:
+            - . publish-test-results
+            - artifact push workflow target/test-results
+
+  - name: Release
+    dependencies: ["Test"]
+    run:
+      when: "branch = 'master' or branch =~ '[0-9]+\\.[0-9]+\\.x'"
```
[the remark this answers is not in the document]
Comment (akanimesh7) [reply .semaphore/semaphore.yml:55]: Right now we are not focusing on the release.. so didn't add it there. But good catch. Updated.

File: .semaphore/semaphore.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+queue:
+  - when: "branch != 'master' and branch !~ '[0-9]+\\.[0-9]+\\.x' and branch !~ 'v\\d+\\.\\d+\\.\\d+\\-hotfix\\-x'"
+    processing: parallel
+
+global_job_config:
+  prologue:
+    commands:
+      - checkout
+      - sem-version java 11
```
[the remark this answers is not in the document]
Comment (akanimesh7) [reply .semaphore/semaphore.yml:27]: No 11 is working fine ! and 11 is what is there in `.java-version` file or imposed by maven-enforcer plugin.

**Review by akanimesh7 — COMMENTED** [review]
Addressed

**Pull request merged**